pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -182,6 +237,36 @@ func getCrashDaemonContainer(cephCluster cephv1.CephCluster, cephVersion version
 	return container
 }
 
+func getCrashPruneContainer(cephCluster cephv1.CephCluster, cephVersion version.CephVersion) corev1.Container {
+	cephImage := cephCluster.Spec.CephVersion.Image
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+	envVars := controller.DaemonEnvVars(cephImage)
+	// Because there is no capability restricting to only "crash prune", this pod
+	// will have admin priviledges.
+	volumeMounts := controller.CephVolumeMounts(dataPathMap, false)
+
+	container := corev1.Container{
+		Name: "ceph-crash-pruner",
+		Command: []string{
+			"ceph",
+			"-c",
+			"/var/lib/rook/rook-ceph/rook-ceph.config",
```

Comment:
Where do you generate the config file?
If you just use `generateCrashEnvVar()` in `envVars` it will just work without the `-c ...`.

Reply:
client.crash doesn't have the capability to run `ceph prune`, so I used this config in the rook_data mount because it had the admin keyring. But on second thought I could just change the client.crash capabilities instead. The change will be in the next push.